pykrige/test.py

```diff
@@ -1388,9 +1388,59 @@ def test_uk3d_functional_drift(self):
         self.assertTrue(np.allclose(ss_func, ss_lin))
 
     def test_geometric_code(self):
+        # NOTE: This test code uses precalculated reference values. If numpy
+        #       RNG changes in future versions the generated coordinate pairs
+        #       might change as well so that reference values need to be
+        #       recalculated.
+        
+        # Create points uniformly distributed across the sphere:
+        np.random.seed(2022017)
+        N=200
+        lon = 360.0*np.random.random(N)
+        lat = 180.0/np.pi*np.arcsin(2.0*np.random.random(N)-1)
+        
+        # For the points generated with this reference seed, the distance matrix
+        # has been calculated using geopy (v. 1.11.0) as follows:
+        # >>>   from geopy.distance import great_circle
+        # >>>   g = great_circle(radius=1.0)
+        # >>>   d = np.zeros((N,N), dtype=float)
+        # >>>   for i in range(N):
+        # >>>       for j in range(N):
+        # >>>           d[i,j] = g.measure((lat[i],lon[i]),(lat[j],lon[j]))
+        # >>>   d *= 180.0/np.pi
+        # From that distance matrix, the reference values have been obtained.
+        
+        # Calculate distance matrix using the PyKrige code:
+        d = np.zeros((N,N))
+        for i in range(N):
+            for j in range(N):
+                d[i,j] = core.great_circle_distance(lon[i],lat[i],lon[j],lat[j])
+        
+        # Test agains reference values:
+        np.testing.assert_allclose([d.max(), d[d>0.0].min(), d.sum(),
+                                    d.std(), np.median(d),
+                                    np.percentile(d, 0.7), d[17,133],
+                                    d[83,33], d[2,76], d[17,45]],
+                                   [179.39827328486047, # max
+                                    0.49781321229536524, # min != 0
+                                    3577568.5670485105, # sum
+                                    39.42686179130957, # std
+                                    90.011132985696634, # median
+                                    5.2851101415176212, # percentile
+                                    106.42435552834375,
+                                    75.241117692324522,
+                                    109.48819700842446,
+                                    40.639971012523361])
```

Comment:
Aww, right is it a `NxN` matrix.. Well we can just use `N=2` then? I mean this is just to check that the output is not off by orders of magnitude, so even checking on a single value could be enough...